mllib/src/main/scala/org/apache/spark/ml/regression/DecisionTreeRegressor.scala

```diff
@@ -88,17 +88,30 @@ final class DecisionTreeRegressor @Since("1.4.0") (@Since("1.4.0") override val
       MetadataUtils.getCategoricalFeatures(dataset.schema($(featuresCol)))
     val oldDataset: RDD[LabeledPoint] = extractLabeledPoints(dataset)
     val strategy = getOldStrategy(categoricalFeatures)
+
+    val instr = Instrumentation.create(this, oldDataset)
+    instr.logParams(params: _*)
+
     val trees = RandomForest.run(oldDataset, strategy, numTrees = 1, featureSubsetStrategy = "all",
-      seed = $(seed), parentUID = Some(uid))
-    trees.head.asInstanceOf[DecisionTreeRegressionModel]
+      seed = $(seed), instr, parentUID = Some(uid))
```

Comment:
I suggest you give the name of the parameter when you have a list of named arguments. In this case, it does not change the meaning of the code, but in some cases with default arguments, this may become ambiguous if we decide to add more arguments.